pkg/clusteragent/orchestrator/transformer.go

```diff
@@ -194,3 +198,181 @@ func extractDeploymentConditionMessage(conditions []v1.DeploymentCondition) stri
 	}
 	return ""
 }
+
+func extractNode(n *corev1.Node) *model.Node {
+	msg := &model.Node{
+		Metadata:      orchestrator.ExtractMetadata(&n.ObjectMeta),
+		PodCIDR:       n.Spec.PodCIDR,
+		PodCIDRs:      n.Spec.PodCIDRs,
+		ProviderID:    n.Spec.ProviderID,
+		Unschedulable: n.Spec.Unschedulable,
+		Status: &model.NodeStatus{
+			Allocatable: map[string]int64{},
+			Capacity:    map[string]int64{},
+		},
+	}
+
+	if len(n.Spec.Taints) > 0 {
+		msg.Taints = extractTaints(n.Spec.Taints)
+	}
+
+	// status
+	msg.Status.KubeletVersion = n.Status.NodeInfo.KubeletVersion
+	msg.Status.ContainerRuntimeVersion = n.Status.NodeInfo.ContainerRuntimeVersion
+	msg.Status.OperatingSystem = n.Status.NodeInfo.OperatingSystem
+	msg.Status.Architecture = n.Status.NodeInfo.Architecture
+	msg.Status.OsImage = n.Status.NodeInfo.OSImage
+	msg.Status.KubeProxyVersion = n.Status.NodeInfo.KubeProxyVersion
+	msg.Status.KernelVersion = n.Status.NodeInfo.KernelVersion
```

Comment:
Sounds like a nit but I would have thought the following would have been easier to read.
Feel free to ignore if that's time sensitive.

```suggestion
	msg := &model.Node{
		Metadata:      orchestrator.ExtractMetadata(&n.ObjectMeta),
		PodCIDR:       n.Spec.PodCIDR,
		PodCIDRs:      n.Spec.PodCIDRs,
		ProviderID:    n.Spec.ProviderID,
		Unschedulable: n.Spec.Unschedulable,
		Status: &model.NodeStatus{
			Allocatable:             map[string]int64{},
			Architecture:            n.Status.NodeInfo.Architecture,
			Capacity:                map[string]int64{},
			ContainerRuntimeVersion: n.Status.NodeInfo.ContainerRuntimeVersion,
			OperatingSystem:         n.Status.NodeInfo.OperatingSystem,
			OsImage:                 n.Status.NodeInfo.OSImage,
			KernelVersion:           n.Status.NodeInfo.KernelVersion,
			KubeletVersion:          n.Status.NodeInfo.KubeletVersion,
			KubeProxyVersion:        n.Status.NodeInfo.KubeProxyVersion,
		},
	}

	if len(n.Spec.Taints) > 0 {
		msg.Taints = extractTaints(n.Spec.Taints)
	}

```